src/java/org/apache/cassandra/io/util/MmappedRegions.java

```diff
@@ -152,15 +152,15 @@ private void updateState(long length)
         }
     }
 
-    private void updateState(CompressionMetadata metadata)
+    private void updateState(CompressionMetadata metadata, long startOffset)
     {
         long offset = 0;
-        long lastSegmentOffset = 0;
+        long lastSegmentOffset = state.startOffset;
```

Comment:
It looks like the code below could easily construct the wrong mapping if `state.startOffset` is not aligned.

Could we assert that it is?